include/deal.II/lac/solver_gmres.h

```diff
@@ -818,34 +827,57 @@ namespace internal
 
 
     template <typename VectorType, typename Enable = void>
-    struct is_dealii_compatible_distributed_vector;
+    struct is_dealii_compatible_vector;
 
     template <typename VectorType>
-    struct is_dealii_compatible_distributed_vector<
+    struct is_dealii_compatible_vector<
       VectorType,
       std::enable_if_t<!internal::is_block_vector<VectorType>>>
     {
-      static constexpr bool value = std::is_same_v<
-        VectorType,
-        LinearAlgebra::distributed::Vector<typename VectorType::value_type,
-                                           MemorySpace::Host>>;
+      static constexpr bool value =
+        std::is_same_v<
+          VectorType,
+          LinearAlgebra::distributed::Vector<typename VectorType::value_type,
+                                             MemorySpace::Host>> ||
+        std::is_same_v<VectorType, Vector<typename VectorType::value_type>>;
     };
 
 
 
     template <typename VectorType>
-    struct is_dealii_compatible_distributed_vector<
+    struct is_dealii_compatible_vector<
       VectorType,
       std::enable_if_t<internal::is_block_vector<VectorType>>>
     {
-      static constexpr bool value = std::is_same_v<
-        typename VectorType::BlockType,
-        LinearAlgebra::distributed::Vector<typename VectorType::value_type,
-                                           MemorySpace::Host>>;
+      static constexpr bool value =
+        std::is_same_v<
+          typename VectorType::BlockType,
+          LinearAlgebra::distributed::Vector<typename VectorType::value_type,
+                                             MemorySpace::Host>> ||
+        std::is_same_v<VectorType, Vector<typename VectorType::value_type>>;
     };
 
 
 
+    template <typename VectorType>
+    MPI_Comm
+    get_mpi_communicator(const VectorType &)
+    {
+      return MPI_COMM_SELF;
+    }
+
+
+
+    template <typename Number>
+    MPI_Comm
+    get_mpi_communicator(
+      const LinearAlgebra::distributed::Vector<Number, MemorySpace::Host> &vec)
+    {
+      return vec.get_mpi_communicator();
+    }
```

Comment:
This is a minor nitpick but could we just add `get_mpi_communicator()` to the vector space vector concept? I think the serial vector is the only vector class which doesn't have this method now.

Reply:
See #17566.